packages/core/src/shared/lsp/utils/platform.ts

```diff
@@ -81,26 +86,121 @@ export async function validateNodeExe(nodePath: string[], lsp: string, args: str
     }
 }
 
+/**
+ * Gets proxy settings and certificates from VS Code
+ */
+export function getVSCodeSettings(): { proxyUrl?: string; certificatePath?: string } {
+    const result: { proxyUrl?: string; certificatePath?: string } = {}
+    const logger = getLogger('amazonqLsp')
+
+    try {
+        // Check if user already has NODE_EXTRA_CA_CERTS set
+        const userCerts = process.env.NODE_EXTRA_CA_CERTS
+        if (userCerts) {
+            logger.info(`User already has NODE_EXTRA_CA_CERTS set: ${userCerts}`)
+            return result
+        }
+
+        // Get proxy settings from VS Code configuration
+        const httpConfig = vscode.workspace.getConfiguration('http')
+        const proxy = httpConfig.get<string>('proxy')
+        if (proxy) {
+            result.proxyUrl = proxy
+            logger.info(`Using proxy from VS Code settings: ${proxy}`)
+        }
+
+        try {
+            // @ts-ignore - we need this function to access certs
+            const certs = tls.getCACertificates()
```

Comment:
not sure how this would work. the `default` behavior is to use the node bundled CAs, which is not what we want. should this be explicitly `system`? or does this just magically work because VSC is patching the return value?